### ISSUE TITLE
Doesn't support quotes in strings

### ISSUE THREAD

**Issue by petetastic** [maxkomarychev/react-native-ultimate-config#22]
This library shows promise, but it doesn't support quoted strings.

e.g. my config file is:

`HELLO: "Here is a test with a \"quote\" inside"`

But in ConfigValues.h that gets compiled to:

`#define HELLO @"Here is a test with a "quote" inside"`

.. which fails to compile in XCode. Any chance of a workaround?

**Comment by maxkomarychev** [comment]
@petetastic hey, I am sorry you have troubles with my library. As a workaround I can suggest to define your strings with single quotes, this way when loaded via `js-yaml` backslashes are properly captured.

```yaml
HELLO: 'Here is a test with a \"quote\" inside'
```

this will properly preserve backslashes when rendered into header file.

you can play with it in online demo: https://nodeca.github.io/js-yaml/

try pasting:

```yaml
HELLO: "Here is a test with a \"quote\" inside"
HELLO2: 'Here is a test with a \"quote\" inside'
```

I am not yet sure whether this is a bug or expected behavior of js-yaml and I'll see what's the best way to solve this. For now please use workaround.

**Comment by maxkomarychev** [comment]
it's a bit more complicated than that.
since ios code is working through `js-yaml` but android is reading `yaml` directly the workaround that works for both platforms (for now) is this:

```
HELLO5:
  ios: 'Here is a test with a \"quote\" inside'
  android: 'Here is a test with a \\\"quote\\\" inside'
```

note android version should have escaped slashes too. stay tuned

**Comment by petetastic** [comment]
Thanks for getting back to me so quickly. That bodes well for this repo :)

Unfortunately my config strings also have single quotes in them (words like "don't" and "can't") which means that the single quote solution wouldn't work.

What I've ended up doing, for now as a temporary solution is to use the backtick (  `  ) operator instead of quotes, then preg_replace'ing them on the client side. Not ideal, but it will do for now. I came here because react-native-config is in woeful shape, so I'm really glad you put this together. Thanks!
